model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -724,80 +715,50 @@ def run_predictor_step(
-        )  # TODO: check
-        end_vertex_local_minus1 = self.grid.get_end_index(
-            dims.VertexDim, h_grid.HorizontalMarkerIndex.local(dims.VertexDim) - 1
-        )
+        start_cell_lb = self.grid.start_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
+        end_cell_lb_level_4 = self.grid.end_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_4))
 
-        start_cell_lb = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim)
-        )
-        end_cell_nudging_minus1 = self.grid.get_end_index(
-            dims.CellDim,
-            h_grid.HorizontalMarkerIndex.nudging(dims.CellDim) - 1,
+        start_edge_lb_level_7 = self.grid.start_index(
+            edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_7)
         )
+        end_edge_halo = self.grid.end_index(edge_domain(h_grid.Zone.HALO))
+        end_edge_local = self.grid.end_index(edge_domain(h_grid.Zone.LOCAL))
 
-        start_edge_lb_plus6 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim) + 6
-        )
-        end_edge_local_minus1 = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim) - 1
-        )
-        end_edge_local = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim)
-        )
+        start_edge_nudging_level_2 = self.grid.start_index(edge_domain(h_grid.Zone.NUDGING_LEVEL_2))
+        end_edge_end = self.grid.end_index(edge_domain(h_grid.Zone.END))
 
-        start_edge_nudging_plus1 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.nudging(dims.EdgeDim) + 1
-        )
-        end_edge_end = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.end(dims.EdgeDim)
-        )
+        start_edge_lb = self.grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY))
 
-        start_edge_lb = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim)
-        )
-        end_edge_nudging = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.nudging(dims.EdgeDim)
-        )
+        end_edge_nudging = self.grid.end_index(edge_domain(h_grid.Zone.NUDGING))
 
-        start_edge_lb_plus4 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim) + 4
-        )
-        start_edge_local_minus2 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim) - 2
-        )
-        end_edge_local_minus2 = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim) - 2
+        start_edge_lb_level_5 = self.grid.start_index(
+            edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_5)
         )
+        edge_second_halo = edge_domain(h_grid.Zone.HALO_LEVEL_2)
```

Comment:
More consistency renaming
```suggestion
        edge_halo_level_2 = edge_domain(h_grid.Zone.HALO_LEVEL_2)
```